applications/services/gui/view_dispatcher.c

```diff
@@ -326,6 +336,44 @@ void view_dispatcher_handle_custom_event(ViewDispatcher* view_dispatcher, uint32
     }
 }
 
+static bool view_dispatcher_handle_signal_exit(ViewDispatcher* instance) {
+    bool is_consumed = false;
+
+    if(instance->event_loop) {
+        furi_event_loop_stop(instance->event_loop);
+        is_consumed = true;
+    }
+
+    return is_consumed;
+}
+
+bool view_dispatcher_handle_signal_event(uint32_t signal, void* arg, void* context) {
```

Comment:
Kinda breaks contract between view dispatcher and application. Should be delivered through the message queue.

Reply:
Also I don't think that view dispatcher should be responsible for signal delivery. User can override callback directly in thread api

Reply:
However, with the current change, how do apps register for signals while also keeping the default View Dispatcher behaviour?

Reply:
@CookiePLMonster keep in mind that this PR is still WIP.

Here is what I imagine in my head:
- Thread signal subscription is going to be low level API, you can use it in applications without EventLoop
- Otherwise you use event loop signals API, which is going to arrive a little bit later

I'll make one more iteration over this PR on this weekend, and then call you for final review

Reply:
@skotopes Fair enough, the PR is not drafted so I considered it final-ish 😁

Reply:
Check now. There will be more polishing in future, but on concept level it's what I wanted.

Reply:
Looks good - and I assume that `furi_event_loop_signal_callback` will propagate the signal down to user-registered signals via the upcoming signals API, right?

Reply:
Yep.